src/DefaultProfile.cpp

```diff
@@ -183,15 +183,18 @@ extern "C"
     int geopm_tprof_init(uint32_t num_work_unit)
     {
         int err = 0;
-        if (g_pmpi_tprof_enabled) {
-            try {
-                int cpu = geopm::Profile::get_cpu();
-                geopm::Profile::default_profile().thread_init(cpu, num_work_unit);
+        try {
+            auto &prof = geopm::Profile::default_profile();
+            if (g_pmpi_tprof_enabled) {
+                prof.thread_init(geopm::Profile::get_cpu());
             }
-            catch (...) {
-                err = geopm::exception_handler(std::current_exception());
+            if (g_pmpi_prof_enabled) {
+                prof.thread_work(num_work_unit);
```

Comment:
Don't the non-lead threads pass 0 here?

Reply:
Only the lead thread has "g_pmpi_prof_enabled", and the non-lead threads have "g_pmpi_tprof_enabled".  I settled on this solution so that the lead thread is responsible for setting the number of work units, but each thread is responsible for zeroing its own counter.  This avoids an error where the lead thread would zero the counter after the non-lead thread had already started incrementing it.  There should be a comment here explaining that!